pkg/agent/ipassigner/linkdetector/link_detector_linux.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkdetector
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/set"
+
+	utilnetlink "antrea.io/antrea/pkg/agent/util/netlink"
+)
+
+var linkSubscribeFunc = netlink.LinkSubscribeWithOptions
+
+type linkDetector struct {
+	mutex         sync.RWMutex
+	cacheSynced   bool
+	eventHandlers map[string][]LinkEventHandler
+	linkNames     set.Set[string]  // known link names
+	linkIndexMap  map[int32]string // map from link index to link name
+	netlink       utilnetlink.Interface
+}
+
+func NewLinkDetector() *linkDetector {
+	return &linkDetector{
+		eventHandlers: make(map[string][]LinkEventHandler),
+		linkNames:     make(set.Set[string]),
```

Comment:
It's better to use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#New, as `make` relies on the implementation detail that Set is just a built-in `map`.